### PR TITLE
obs-studio-plugins.obs-backgroundremoval: 0.5.16 -> 1.1.13

### PR DESCRIPTION
**Pull request by Zahrun**
###### Description of changes

https://github.com/royshil/obs-backgroundremoval/compare/v0.5.16...1.1.13

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Requires https://github.com/NixOS/nixpkgs/issues/228093 (includes tensorrt_provider_factory.h to build)

### CODE REVIEW

**Comment by Zahrun** [comment]
Still pending https://github.com/NixOS/nixpkgs/issues/228093

**Comment by Zahrun** [comment]
https://github.com/NixOS/nixpkgs/pull/260745 seems promising

**Comment by spikespaz** [comment]
I had trouble finding the most recent version of the plugin that builds with onnx 1.15.1 (current in unstable). I've been trying for well over an hour.

**Comment by Zahrun** [comment]
Maybe we could ask the plugin developers to provide a way to build it in CPU-only mode, which would not require tensorrt_provider_factory.h to build, if that is not already an option.

**Comment by Zahrun** [comment]
See https://github.com/occ-ai/obs-backgroundremoval/issues/523

**Comment by Zahrun** [comment]
This now builds correctly without GPU support on nix stable.
On master, we have an issue to build onnx which blocks us https://github.com/NixOS/nixpkgs/pull/281011

**Comment by Zahrun** [comment]
onnx seems fine. now onnxruntime build is also broken. should be fixed by https://github.com/NixOS/nixpkgs/pull/258392 according to https://github.com/NixOS/nixpkgs/pull/281011#issuecomment-1894183833

**Comment by Zahrun** [comment]
Ok this builds now.
However it does not work yet.
It seems the plugin is not finding the model path.
`info: [obs-backgroundremoval]   Model file path: (null)`

**Comment by Zahrun** [comment]
@spikespaz I opened an issue on the developers’ repo at https://github.com/occ-ai/obs-backgroundremoval/issues/544.
This is probably more a Nix packaging issue but I’m not sure exactly what’s going on. If you have an idea, please suggest (or anyone else’s help is welcome of course)

**Comment by Zahrun** [comment]
Fixed the installation path by removing the extra /usr in the install cmake line.
Also updated to last release version.
Ready for review.

File: pkgs/applications/video/obs-studio/plugins/obs-backgroundremoval/default.nix [hunk 1]
```diff
@@ -2,44 +2,52 @@
...
-  version = "0.5.16";
+  version = "1.1.13";
 
   src = fetchFromGitHub {
-    owner = "royshil";
+    owner = "occ-ai";
     repo = "obs-backgroundremoval";
-    rev = "v${version}";
-    hash = "sha256-E+pm/Ma6dZTYlX3DpB49ynTETsRS2TBqgHSCijl/Txc=";
+    rev = "${version}";
```
Comment (SuperSandro2000) [inline pkgs/applications/video/obs-studio/plugins/obs-backgroundremoval/default.nix:20]:
```suggestion
    rev = version;
```

File: pkgs/applications/video/obs-studio/plugins/obs-backgroundremoval/default.nix [hunk 2]
```diff
@@ -2,44 +2,52 @@
...
-    rm -rf $out/obs-plugins
-    mv $out/data $out/share/obs
+  buildPhase = ''
+     cd ..
+     cmake --build build_x86_64 --parallel
+  '';
+
+  installPhase = ''
+    cmake --install build_x86_64 --prefix "$out"
   '';
```
Comment (SuperSandro2000) [inline pkgs/applications/video/obs-studio/plugins/obs-backgroundremoval/default.nix:44]: Can't we use cmakeFlags and the normal phases?

File: pkgs/applications/video/obs-studio/plugins/obs-backgroundremoval/default.nix [hunk 3]
```diff
@@ -2,44 +2,52 @@
...
-  nativeBuildInputs = [ cmake ];
-  buildInputs = [ obs-studio onnxruntime opencv ];
+  nativeBuildInputs = [ cmake ninja ];
+  buildInputs = [ obs-studio onnxruntime opencv qt6.qtbase curl ];
 
   dontWrapQtApps = true;
 
   cmakeFlags = [
+    "--preset linux-x86_64"
+    "-DCMAKE_MODULE_PATH:PATH=/build/source/cmake"
```
Comment (SuperSandro2000) [inline pkgs/applications/video/obs-studio/plugins/obs-backgroundremoval/default.nix:31]: We should use a relative path here

**Comment by spikespaz** [comment]
Is this PR still relevant or do we need a new one?

File: pkgs/applications/video/obs-studio/plugins/obs-backgroundremoval/default.nix (hunk 2 above)
Comment (Zahrun) [reply pkgs/applications/video/obs-studio/plugins/obs-backgroundremoval/default.nix:44]: cmakeFlags will apply to the normal buildPhase, right?
How to specify flags for installPhase?
Where is the documentation about this?

**Comment by Zahrun** [comment]
> Is this PR still relevant or do we need a new one?

Sorry I forgot this was pending. In my head this was done and merged. Let me check the review comments.

File: pkgs/applications/video/obs-studio/plugins/obs-backgroundremoval/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/applications/video/obs-studio/plugins/obs-backgroundremoval/default.nix:44]: I think the project is using pretty new cmake stuff we don't have explicit support for.

---


> cmakeFlags will apply to the normal buildPhase, right?

It applies to the configure phase https://github.com/NixOS/nixpkgs/blob/c5b25a6eb94539cdeac8971e05f71e12e9e7e8a3/pkgs/by-name/cm/cmake/setup-hook.sh#L141-L144

> How to specify flags for installPhase?

Apparently there is no hook for that.

>  Where is the documentation about this?

https://github.com/NixOS/nixpkgs/blob/master/doc/hooks/cmake.section.md

Comment (Zahrun) [reply pkgs/applications/video/obs-studio/plugins/obs-backgroundremoval/default.nix:44]: Ok

**Comment by spikespaz** [comment]
I forgot to say thank you, @Zahrun and others.

File: pkgs/applications/video/obs-studio/plugins/obs-backgroundremoval/default.nix [hunk 4]
```diff
@@ -2,44 +2,52 @@
...
 }:
 
 stdenv.mkDerivation rec {
   pname = "obs-backgroundremoval";
-  version = "0.5.16";
+  version = "1.1.13";
 
   src = fetchFromGitHub {
-    owner = "royshil";
+    owner = "occ-ai";
```
Comment (Ten0) [inline pkgs/applications/video/obs-studio/plugins/obs-backgroundremoval/default.nix:18]: It's unclear to me why this change happens here.
It seems that that redirects to `royshil/obs-backgroundremoval` (ATM) but I don't understand why we're using this instead of directly the source repo.

Comment (spikespaz) [reply pkgs/applications/video/obs-studio/plugins/obs-backgroundremoval/default.nix:18]: I think I remember the repository at one point being owned by @occ-ai, and it has since transferred ownership back to @royshil
